lib/cast/double.js

```diff
@@ -14,21 +14,17 @@ const BSON = require('bson');
  */
 
 module.exports = function castDouble(val) {
-  if (val == null) {
-    return val;
-  }
-  if (val === '') {
+  if (val == null || val === '') {
     return null;
   }
 
   let coercedVal;
-  if (val instanceof BSON.Int32 || val instanceof BSON.Double) {
-    coercedVal = val.value;
-  } else if (val instanceof BSON.Long) {
+  if (val instanceof BSON.Long) {
```

Comment:
Can we use the same logic we chose to use in the Int32 PR here as well?

Reply:
Good catch!